src/app/questionGenerator.js

```diff
@@ -0,0 +1,44 @@
+class QuestionGenerator {
+
+    //argument argAmountOfId will be relaced by connection to CurrentMode and it's IDs'
+    static generateQuestionsForCurrentMode(amountOfId) {
+        const amountOfAnswersPerQuestion = 4;
+        let generatedIds = [];
+
+        if (amountOfId < amountOfAnswersPerQuestion) {
+            throw `amountOfId must be greater or equal ${amountOfAnswersPerQuestion}. Passed value: ${amountOfId}`
+        }
+
+        for (let index = 0; index < amountOfAnswersPerQuestion; index++) {
+            let id;
+            do {
+                id = generateRandomId(amountOfId)
+            }
+            while (generatedIds.includes(id) === true);
+        
+            generatedIds.push(id);
+        
+        }
+        return generatedIds;
+    }
+
+    static chooseCorrectAnswer(arrayWithId = []) {
+        const minIndex = 0;
+        const maxIndex = 3;
+        const index = Math.floor(Math.random() * (maxIndex - minIndex + 1) + minIndex);
+        const rightAnswer = arrayWithId[index];
+        return rightAnswer;
+        
+    }
+   
+}
+
+
+// ------------------PRIVATE FUNCTIONS----------------------------------------
+
+
+function generateRandomId(amountOfId) {
```

Comment:
Can you give me more of your thoughts about why this function is outside of the class?

Reply:
I wanted to make it "private", not seen from outside. I want to export default just class QuestionGenerator. So QuestionGenerator's functions can use that function, but from other files it is impossible to use generateRandomId.
Then we don't have a problem that argument "amountOfId" is different than number, because it is taken from another function (generateQuestionsForCurrentMode). 
Now it is primitive, but in the future I want to replace it from outside, and it's value would be lenght of array with objects from JSON

Reply:
JS is a bit confusing when it comes to private/protected/public, as all class fields are by default public. This approach does the trick, but in my opinion it mixes the oop with functional programming. Some further reading if you are interested: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_class_fields
https://css-tricks.com/implementing-private-variables-in-javascript/